packages/api/src/useFetch.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+import { getQueryKeys, send } from './utils';
+import type {
+    TSocketEndpointNames,
+    TSocketAcceptableProps,
+    TSocketResponseData,
+    TSocketRequestCleaned,
+    TSocketRequestQueryOptions,
+} from '../types';
+
+const useFetch = <T extends TSocketEndpointNames>(name: T, ...props: TSocketAcceptableProps<T, true>) => {
+    const prop = props?.[0];
+    const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestCleaned<T>) : undefined;
```

Comment:
in case of `prop` or `payload` or `options` is empty then it will return false in them. shouldn't you check for the value inside them and then return them in useQuery?

Reply:
@ali-hosseini-deriv It won't be `false`, It will be `undefined` which is being handled in both the `getQueryKeys` and the `send` methods.